acceptance-tests/src/test/java/linea/plugin/acc/test/rpc/linea/SetExtraDataTest.java

```diff
@@ -0,0 +1,264 @@
+
+    final var reqLinea = new FailingLineaSetExtraDataRequest(tooShortExtraData);
+    final var respLinea = reqLinea.execute(minerNode.nodeRequests());
+    assertThat(respLinea.getMessage()).isEqualTo("Expected 32 bytes but got 31");
+  }
+
+  @Test
+  public void successfulSetExtraData() {
+    final var extraData =
+        Bytes32.fromHexString("0x0100000000000000000000000000000000000000000000000000000000000000");
+
+    final var reqLinea = new LineaSetExtraDataRequest(extraData);
+    final var respLinea = reqLinea.execute(minerNode.nodeRequests());
+    assertThat(respLinea).isTrue();
+  }
+
+  @Test
+  public void successfulUpdateMinGasPrice() {
+    final var doubledMinGasPriceKWei = MIN_GAS_PRICE.multiply(2).divide(1000);
+    final var hexMinGasPrice =
+        Strings.padStart(doubledMinGasPriceKWei.toShortHexString().substring(2), 8, '0');
+    final var extraData =
+        Bytes32.fromHexString(
+            "0x010000000000000000" + hexMinGasPrice + "00000000000000000000000000000000000000");
+
+    final var reqLinea = new LineaSetExtraDataRequest(extraData);
+    final var respLinea = reqLinea.execute(minerNode.nodeRequests());
+    assertThat(respLinea).isTrue();
+    assertThat(minerNode.getMiningParameters().getMinTransactionGasPrice())
+        .isEqualTo(MIN_GAS_PRICE.multiply(2));
+  }
+
+  @Test
+  public void successfulUpdatePricingParameters() throws IOException {
+    final Web3j web3j = minerNode.nodeRequests().eth();
+    final Credentials credentials = Credentials.create(Accounts.GENESIS_ACCOUNT_ONE_PRIVATE_KEY);
+    final TransactionManager txManager = new RawTransactionManager(web3j, credentials, CHAIN_ID);
+
+    final KeccakDigest keccakDigest = new KeccakDigest(256);
+    final StringBuilder txData = new StringBuilder();
+    txData.append("0x");
+    for (int i = 0; i < 10; i++) {
+      keccakDigest.update(new byte[] {(byte) i}, 0, 1);
+      final byte[] out = new byte[32];
+      keccakDigest.doFinal(out, 0);
+      txData.append(new BigInteger(out));
+    }
+
+    final var txUnprofitable =
+        txManager.sendTransaction(
+            MIN_GAS_PRICE.getAsBigInteger(),
+            BigInteger.valueOf(MAX_TX_GAS_LIMIT / 2),
+            credentials.getAddress(),
+            txData.toString(),
+            BigInteger.ZERO);
+
+    final Account sender = accounts.getSecondaryBenefactor();
+    final Account recipient = accounts.createAccount("recipient");
+    final TransferTransaction transferTx = accountTransactions.createTransfer(sender, recipient, 1);
+    final var txHash = minerNode.execute(transferTx);
+
+    minerNode.verify(eth.expectSuccessfulTransactionReceipt(txHash.toHexString()));
+
+    // assert that tx below margin is not confirmed
+    minerNode.verify(eth.expectNoTransactionReceipt(txUnprofitable.getTransactionHash()));
+
+    final var zeroFixedCostKWei = "00000000";
+    final var minimalVaribleCostKWei = "00000001";
+    final var minimalMinGasPriceKWei = "00000001";
```

Comment:
Small issue with this approach is that if by any chance the implementation confuses `minimalVaribleCostKWei ` and `minimalMinGasPriceKWei` or duplicates the value of either of the two, we won't catch it with this expected result. Can we set them to any not matching values?

Reply:
good point, will change